hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java

```diff
@@ -1071,6 +1096,12 @@ static BalancerParameters parse(String[] args) {
               b.setSortTopNodes(true);
               LOG.info("Balancer will sort nodes by" +
                   " capacity usage percentage to prioritize top used nodes");
+            } else if ("-limitOverUtilizedNum".equalsIgnoreCase(args[i])) {
+              Preconditions.checkArgument(++i < args.length,
+                  "limitOverUtilizedNum value is missing: args = " + Arrays.toString(args));
+              int limitNum = Integer.parseInt(args[i]);
```

Comment:
Here we need to check if `limitNum` is less than 1?

Reply:
Fixed, 0 is legal, non-negative numbers are okay.

Reply:
Now, restricting the list that is greater than the threshold, and in the future, we will also need to restrict the list that is greater than the average (aboveAvgUtilized).